lib/instructions/instructions.c

```diff
@@ -0,0 +1,1602 @@
+/* EVMC: Ethereum Client-VM Connector API.
+ * Copyright 2018 Pawel Bylica.
+ * Licensed under the MIT License. See the LICENSE file.
+ */
+
+#include <evmc/instructions.h>
+
+/** Marks an instruction as undefined. */
+#define UNDEFINED -1
+
+/**
+ * Gas price tiers, names from Yellow Paper.
+ * @{
+ */
+#define ZERO 0
+#define BASE 2
+#define VERYLOW 3
+#define LOW 5
+#define MID 8
+#define HIGH 10
+
+/** @} */
+
+static struct evmc_instruction_metrics constantinople_metrics[256] = {
```

Comment:
This file is 1600 lines, maybe put revisions in separate files? It would be also easier to diff them

Reply:
After thinking about this for 2 day, I'd like to keep it as it is. I hope this is single time job and it should depend on testing, not manual review (some parts already tested, I can add test cases for all HFs).

1. To add a new EVM revision you will have to copy the recent one and apply changes. This should not be very complicated. In case it was in separate files, you will also have to alter the CMake config.

2. Having all this in 2 files make it easier to copy to other projects, even translate to other languages.

3. Having the tables in separate files require to make the symbol default visibility.